README.md

```diff
@@ -1,3 +1,110 @@
 # Spend Permissions
 
+> :information_source: These contracts are unaudited. Please use at your own risk.
+
 **Spend Permissions enable apps to spend native and ERC-20 tokens on behalf of users.**
+
+## Design Overview
+
+### 1. Periphery addition to Coinbase Smart Wallet V1
+
+While implementing this feature as a new V2 wallet implementation was tempting, we decided to leverage the modular owner system from [Smart Wallet V1](https://github.com/coinbase/smart-wallet) and avoid a hard upgrade.
+
+### 2. Only Native and ERC-20 token support
+
+Spend Permissions only supports spending Native (e.g. ETH) and ERC-20 (e.g. USDC) tokens on a recurring period. This enables use cases like subscriptions out of the box (e.g 10 USDC per month) and also can support apps that want to avoid asking users for spend permissions every session.
+
+This approach does **not** enable apps to make arbitrary external calls from user accounts, improving security by having a tighter and fully-known scope of account control.
+
+### 3. Spender-originated calls
+
+Spend Permissions allow users to delegate token spending to a `spender` address, presumably controlled by the app. When an app wants to spend user tokens, it calls into `SpendPermissionManager` from this `spender` address. `SpendPermissionManager` will then validate the spend is within the approved permission's allowance and calls into the user's account to transfer tokens.
+
+This approach does **not** use the ERC-4337 EntryPoint to prompt external calls from user accounts, improving security by avoiding the possibility of ERC-4337 Paymasters spending users' tokens on gas fees.
```

Comment:
and above perhaps instead of "improving security" could just read ".... from user accounts, maintaining a tight and fully-known scope of account control."